### PR TITLE
refactor: use `xrpl-codec-gen` for `definitions.json`

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

This PR uses [xrpl-codec-gen](https://github.com/RichardAH/xrpl-codec-gen) to generate the `definitions.json` file. This rearranges the file a bit. It also removes some legacy bits of code that aren't actually in the rippled source code. 

No change in functionality.

### Context of Change

@RichardAH mentioned this repo to me.

### Type of Change

- [x] Refactor (non-breaking change that only restructures code)

## Test Plan

Tests pass. A couple of the data-driven tests needed to be removed because they weren't actually in the source code.

### CODE REVIEW

File: xrpl/core/binarycodec/definitions/definitions.json [hunk 1]
```diff
@@ -1,29 +1,32 @@
...
   "TYPES": {
-    "Validation": 10003,
     "Done": -1,
+    "Unknown": -2,
+    "NotPresent": 0,
+    "UInt16": 1,
+    "UInt32": 2,
+    "UInt64": 3,
     "Hash128": 4,
+    "Hash256": 5,
```
[the remark this answers is not in the document]
Comment (mvadari) [reply xrpl/core/binarycodec/definitions/definitions.json:10]: I believe that's what it's called in rippled.

[the remark this answers is not in the document]
Comment (dangell7) [reply xrpl/core/binarycodec/definitions/definitions.json:10]: Actually its the other way around.. I might have messed that part up but when I looked in the current definitions file in the py/js repos it said `Hash256` and the output from rippled is UInt.

https://github.com/RichardAH/xrpl-codec-gen/blob/8e938ee433a89b5ffd951ca02787185bda0bb64d/gen.js#L26

[the remark this answers is not in the document]
Comment (mvadari) [reply xrpl/core/binarycodec/definitions/definitions.json:10]: We can certainly change it back to match rippled, but IMO that's out of the scope of this PR.

**Comment by mvadari** [comment]
diff is here: https://github.com/XRPLF/xrpl.js/pull/2037#issuecomment-1188937970

**Review by JST5000 — APPROVED** [review]
LGTM

**Review by ckniffen — APPROVED** [review]

**Pull request merged**